Implement migration survey for WOOSHIP-1511

Ferdev opened the pull request:
## Description

  This PR implements WOOSHIP-1511: a migration survey to understand why WCS&T users haven't migrated to WooCommerce Shipping. The survey appears after users complete a shipping label
  purchase to capture feedback on migration blockers.

  ### Related issue(s)

  Implements WOOSHIP-1511: https://linear.app/a8c/issue/WOOSHIP-1511/implement-survey-in-wcsandt-for-migration-blockers

  ### Key Features Implemented

  - **Backend Survey Logic**: PHP class handling survey eligibility, display limits, and tracking
  - **Frontend Modal Component**: React modal with primary/followup questions matching Linear spec
  - **Post-Purchase Timing**: Survey displays after label purchase completion (not during)
  - **Display Control**: Maximum 2 displays per user with 3-day cooldown period
  - **Analytics Integration**: Tracks survey events via WooCommerce Services Tracks
  - **Responsive Design**: Mobile-friendly modal with proper styling

  ### Survey Flow

  1. User completes shipping label purchase
  2. Survey appears after purchase dialog closes (500ms delay for print dialogs)
  3. Primary question with 7 predefined options
  4. Conditional followup textarea based on selection
  5. Submit or skip options with proper tracking
  6. Confirmation message with migration guidance

  ### Display Logic

  - **Eligibility**: Only shown if WooCommerce Shipping is not active
  - **Frequency**: Maximum 2 times per user with 3-day cooldown
  - **Completion**: Never shows again after user submits feedback
  - **Testing**: `?force_survey=1` parameter for development/testing

  ### Steps to reproduce & screenshots/GIFs

  **Testing Steps:**
  1. Go to WP Admin → Orders → Edit any order
  2. Add `?force_survey=1` to URL to force survey display
  3. Create a shipping label (or use existing label creation flow)
  4. Complete the purchase process
  5. Survey modal should appear after purchase completion
  6. Test both primary questions and followup responses
  7. Verify tracking in browser dev tools (AJAX calls to track display/submission)

  **Track Events to Monitor:**

  When testing, verify these events are properly fired in browser dev tools (Network tab → XHR requests) or analytics dashboard:

  1. **`migration_survey_survey_displayed`**
     - Fired when: Survey modal appears to user
     - Properties: None
     - Expected: Once per survey display

  2. **`migration_survey_survey_submitted`**
     - Fired when: User completes and submits survey
     - Properties:
       - `primary_reason`: Selected option from dropdown
       - `followup_*`: Any followup question responses (prefixed with `followup_`)
     - Expected: Once per successful submission

  3. **`migration_survey_survey_dismissed`**
     - Fired when: User clicks "Skip" or closes survey
     - Properties: None
     - Expected: Once per dismissal action

  ### Checklist

  - [ ] unit tests
  - [x] `changelog.txt` entry added
  - [x] `readme.txt` entry added

claude committed:
Add migration survey backend implementation

Implements WOOSHIP-1511 backend functionality:
- WC_Connect_Migration_Survey class with display logic
- AJAX handlers for survey submission, dismissal, and display tracking
- User meta tracking for display count and cooldown periods
- Tracks events integration for analytics
- Survey eligibility checks (WooCommerce Shipping not active, display limits)
- WordPress hooks integration for shipping label script localization

claude committed:
Add migration survey frontend components

Implements WOOSHIP-1511 frontend components:
- MigrationSurveyModal React component with primary/followup questions
- Survey styling with proper radio button alignment
- Content matching Linear issue specification exactly
- Modal backdrop and responsive design
- Form validation and submission handling
- Confirmation message after survey submission

claude committed:
Integrate migration survey with shipping label flow

Implements survey timing and integration:
- Show survey after purchase dialog closes (not during purchase)
- Track when purchase dialog was actually shown vs just closed
- 500ms delay to allow print dialogs to close first
- AJAX integration for survey submission and display tracking
- Survey only appears after successful label purchase completion

claude committed:
Add migration survey CSS integration

Import migration survey styles into main WooCommerce Services stylesheet
for proper style loading and integration with existing WCS UI components.

claude committed:
Fix script handle for migration survey data injection

Change wp_localize_script target from 'wc-connect-create-shipping-label'
to 'wc_connect_admin' to ensure survey data reaches the frontend properly.
The shipping label script wasn't the correct target for localization.

claude committed:
Improve migration survey layout and alignment

- Center the question title for better visual balance
- Center radio button + answer blocks with max-width constraint
- Keep answer text left-aligned next to radio buttons
- Increase radio button margin for better spacing
- Ensure textarea labels remain properly aligned

claude committed:
Fix radio button and answer alignment in survey

- Use justify-content: center to center radio+answer as a unit
- Change span from flex: 1 to flex-shrink: 0 to prevent text stretching
- Remove max-width constraint on option container
- Now radio buttons and answers appear as centered, compact units

claude committed:
Use inline-block layout for centered radio button alignment

- Remove flexbox layout in favor of inline-block approach
- Set text-align: center on container to center radio+text pairs
- Use vertical-align: top for proper radio button alignment
- Set max-width on text spans to control line wrapping
- Keep answer text left-aligned within its inline-block container

claude committed:
Simplify radio button layout with proper spacing

- Return to flexbox layout for reliable radio+text alignment
- Add horizontal padding to options container for visual centering
- Remove complex text-align and inline-block approaches
- Radio buttons now properly positioned 8px from their labels
- Options block centered via padding while maintaining left alignment

claude committed:
Fix radio button alignment with specific margin overrides

Address generic CSS rule adding 24px margin-left to spans in labels:
- Add marginLeft: '0 \!important' and margin: '0' to override existing CSS
- Set justifyContent: 'flex-start' to align all options to the left
- Keep 50px padding wrapper for visual centering
- Radio buttons now appear directly next to their labels (8px spacing)

This targets the specific styling issue identified in the WordPress admin CSS.

claude committed:
Remove inline styles and fix radio alignment with proper CSS

- Remove all inline styles from React component
- Add text-align: left to .migration-survey__option
- Add margin-left: 0 \!important to spans to override WordPress admin CSS
- Clean, maintainable solution using proper SCSS styling

This overrides the 24px margin-left from WordPress admin CSS while
keeping radio buttons properly aligned with their labels.

claude committed:
Align all survey modal text to the left

- Add text-align: left to all major modal sections
- Left-align modal header, content, questions, and footer
- Change footer button alignment from flex-end to flex-start
- Ensure consistent left alignment throughout the entire modal
- Remove centered text styling for better readability

This creates a more standard form layout with left-aligned text.

claude committed:
Position Skip button on left and Submit Feedback on right

Change footer button layout from flex-start to space-between:
- Skip button appears on the left side
- Submit Feedback button appears on the right side
- Maintains proper spacing between buttons
- Follows standard modal button conventions

This provides better visual balance and follows UX patterns where
secondary actions (Skip) are on the left and primary actions
(Submit) are on the right.

claude committed:
Fix migration survey style import path

Update the migration survey style import to use the local components path instead of the global path, ensuring the styles are loaded correctly within the woocommerce-services extension.

claude committed:
Add migration survey entry to changelog and readme

- Add entry for migration survey feature in both changelog.txt and readme.txt
- Feature helps understand WooCommerce Shipping adoption blockers

claude committed:
Merge branch 'trunk' into wooship-1511-implement-survey-in-wcst-for-migration-blockers

ismaeldcom commented on woocommerce-services.php:
```diff
@@ -873,9 +873,13 @@ public function load_dependencies() {
 			$shipping_label         = new WC_Connect_Shipping_Label( $api_client, $settings_store, $schemas_store, $payment_methods_store );
 			$nux                    = new WC_Connect_Nux( $tracks, $shipping_label );
-			$store_notices_notifier = new StoreNoticesNotifier( $taxes_logger->is_debug_enabled() );
+			$store_notices_notifier = class_exists( 'Automattic\WCServices\StoreNotices\StoreNoticesNotifier' )
```
Why do we need to check if the class exist? 🤔

ismaeldcom committed:
Merge branch 'trunk' into wooship-1511-implement-survey-in-wcst-for-migration-blockers

ismaeldcom commented on classes/class-wc-connect-shipping-label.php:
```diff
@@ -529,10 +529,6 @@ private function calculate_should_show_meta_box( $post ) {
 		 * @return bool True if shipping label is enabled from the settings.
 		 */
 		public function is_shipping_label_enabled() {
-			// For tax-only installations, shipping labels are disabled
```
Any reason to remove this? 🤔

Ferdev replied:
yeah, this was avoiding old users who were allowed to use the shipping functionality to keep using it. It's a bug introduced in latest version.

ismaeldcom commented on classes/class-wc-connect-migration-survey.php:
```diff
@@ -0,0 +1,195 @@
+		 * Track survey events
+		 */
+		private function track_event( $event_name, $properties = array() ) {
+			if ( class_exists( 'WC_Connect_Tracks' ) && class_exists( 'WC_Connect_Logger' ) && function_exists( 'wc_get_logger' ) ) {
```
This is just an example, but I can see Claude being overly cautious, adding unnecessary checks, and making things far more complex than necessary.

ismaeldcom commented on client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/index.js:
```diff
@@ -26,45 +27,118 @@ import FeatureAnnouncement from 'components/migration/feature-announcement';
+	}, [ loaded, showPurchaseDialog ] );
+
+	// Show survey when purchase dialog is closed (after it was actually shown)
+	React.useEffect( () => {
```
Could we simplify this be just showing the survey using the `onClose` handler?

ismaeldcom commented on client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/index.js:
```diff
@@ -26,45 +27,118 @@ import FeatureAnnouncement from 'components/migration/feature-announcement';
-			<Sidebar siteId={props.siteId} orderId={props.orderId}/>
-		</div>
-	</Modal>
+	const trackSurveyDisplay = async () => {
```
I think `trackSurveyDisplay` and `handleSurveySubmit` should be in the `MigrationSurveyModal` not here.

ismaeldcom commented on classes/class-wc-connect-migration-survey.php:
```diff
@@ -0,0 +1,195 @@
+					$wc_logger = wc_get_logger();
+					$logger = new WC_Connect_Logger( $wc_logger );
+					$tracks = new WC_Connect_Tracks( $logger, WCSERVICES_PLUGIN_FILE );
+					$tracks->record_user_event( 'migration_survey_' . $event_name, $properties );
```
Nitpick: We're including `survey_survey` on every event name.

ismaeldcom reviewed (commented):
Thanks for taking care of this. The solution works, and the survey looks good. However, I think the implementation could be improved.

I'm not sure if we intend to limit our time commitment to this extension as much as possible, but using Claude to take care of it results in _questionable quality_ 🙂

Apart from the review comments, I think it would be simpler to use `recordTracksEvent` on the JS side, rather than making AJAX calls with nonce validation to track events.  Also, the SCSS doesn't use any existing variables or styles.

claude committed:
Address PR comments

Ferdev commented:
Hey @ismaeldcom, I think i've addressed most of your suggestions. I was kind of in a hurry when I implemented this, and with my AFK closing by, didn't have much time to test it properly. I guess somebody else will take care of this while I'm away. But maybe, in the meantime, you can give it another look.

samnajian committed:
Address QIT security concerns

samnajian commented:
@ismaeldcom I've addressed QIT security issues, do you think we can merge this PR as is and create a follow up issue to clean up?

ismaeldcom commented:
>do you think we can merge this PR as is and create a follow up issue to clean up?

We can, although I don't think we should invest more time in something like this.

The code looks better now. However, I still think the display logic is a bit too complicated, and the survey feels off because of the use of custom styles instead of WP/WC components.


https://github.com/user-attachments/assets/ed75f568-d160-4168-af9e-07a1fb7f1dd3

ismaeldcom reviewed (commented):
>do you think we can merge this PR as is and create a follow up issue to clean up?

We can, although I don't think we should invest more time in something like this.

The code looks a bit better now. However, I still think the display logic is a bit too complicated, and the survey feels off because of the use of custom styles instead of WP/WC components.

@samnajian could you confirm that the events are being registered? It was working for me yesterday, but not today, not sure if it's due to my network issues 🤔 


https://github.com/user-attachments/assets/ed75f568-d160-4168-af9e-07a1fb7f1dd3

samnajian commented:
> @samnajian could you confirm that the events are being registered? It was working for me yesterday, but not today, not sure if it's due to my network issues 🤔


@ismaeldcom Only the `display` event triggers a separate tracking ajax call, the rest are tracked at the end of the method that handles the main action request, see https://github.com/Automattic/woocommerce-services/pull/2877/files#diff-57a181d83d0becfa757e1cc48709f7a14c6e9209bdbb6ccc2e9b8889faacbef7R138

ismaeldcom commented:
> Only the display event triggers a separate tracking ajax call, the rest are tracked at the end of the method that handles the main action request

I know, I meant if you're seeing them on [Tracks](https://mc.a8c.com/tracks/live/), because I don't.

samnajian commented:
> I know, I meant if you're seeing them on [Tracks](https://mc.a8c.com/tracks/live/), because I don't.

This is what I see 
<img width="816" height="359" alt="image" src="https://github.com/user-attachments/assets/90c91303-bbc0-41b3-a321-5e9b81155507" />

However the `dismissal` is recorded as `submitted` as well!

samnajian committed:
Remove redundant tracking even prefixes

ismaeldcom commented:
Thanks for checking it. The proxy doesn't work when I'm tethering.

Looks like in the last commit, [skipped became submitted with `primary_reason: skipped`](https://github.com/Automattic/woocommerce-services/blob/9132869a79293ee33439a4b9c005fb9e5e238627/client/components/migration-survey/modal.js#L135).

samnajian committed:
Fix button styling and submit a separate tracks event for dismissal

samnajian committed:
Track modal close as dismiss and add aria label

samnajian commented:
@ismaeldcom I just addressed some concerns, updated the button styling to match the plugin's design and removed redundancy in Tracks even names, please give it another check.
<img width="634" height="817" alt="image" src="https://github.com/user-attachments/assets/7edc5b30-2f44-40c3-a1c8-2c057ec547fd" />
<img width="825" height="352" alt="image" src="https://github.com/user-attachments/assets/63307c26-40c4-4d98-ba60-d0b63c38a714" />

ismaeldcom reviewed (approved):
Now it works as expected, and looks closer to the rest of the UI.
